### PR TITLE
[OSDOCS-6989] Control plane machine sets for clusters on RHOSP

### PR DESCRIPTION
**Pull request by maxwelldb**
Version(s): 4.14

Issue: [OSDOCS-6989](https://issues.redhat.com//browse/OSDOCS-6989)

Link to docs preview:
- [Limitations](https://66212--docspreview.netlify.app/openshift-enterprise/latest/machine_management/control_plane_machine_management/cpmso-about#cpmso-limitations_cpmso-about)
- [Supported cloud providers](https://66212--docspreview.netlify.app/openshift-enterprise/latest/machine_management/control_plane_machine_management/cpmso-getting-started#cpmso-platform-matrix_cpmso-getting-started)
- [Creating a control plane machine set custom resource ](https://66212--docspreview.netlify.app/openshift-enterprise/latest/machine_management/control_plane_machine_management/cpmso-getting-started#cpmso-creating-cr_cpmso-getting-started)
- [Sample YAML for configuring OpenStack clusters](https://66212--docspreview.netlify.app/openshift-enterprise/latest/machine_management/control_plane_machine_management/cpmso-configuration#cpmso-sample-yaml-openstack_cpmso-configuration)
- [Enabling OpenStack features for control plane machines](https://66212--docspreview.netlify.app/openshift-enterprise/latest/machine_management/control_plane_machine_management/cpmso-using#cpmso-supported-features-openstack_cpmso-using)
- [Changing the OpenStack nova flavor by using a control plane machine set](https://66212--docspreview.netlify.app/openshift-enterprise/latest/machine_management/control_plane_machine_management/cpmso-using#cpms-changing-openstack-flavor-type_cpmso-using)
- [Failure domain  platform support and configuration](https://66212--docspreview.netlify.app/openshift-enterprise/latest/machine_management/control_plane_machine_management/cpmso-resiliency#cpmso-failure-domains-provider_cpmso-resiliency)
- [Configuring RHOSP clusters that have machines with root volume availability zones after an upgrade](https://66212--docspreview.netlify.app/openshift-enterprise/latest/machine_management/control_plane_machine_management/cpmso-troubleshooting#cpmso-openstack-ts-az-config_cpmso-troubleshooting)
- [Configuring RHOSP clusters that have control plane machines with availability zones after an upgrade](https://66212--docspreview.netlify.app/openshift-enterprise/latest/machine_management/control_plane_machine_management/cpmso-troubleshooting#cpmso-openstack-with-az-config_cpmso-troubleshooting)

QE review:
- [x] QE has approved this change.

Additional information: This is a continuation of #64454.

### CODE REVIEW

File: modules/cpmso-openstack-with-az-config.adoc [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+$ oc describe controlplanemachineset.machine.openshift.io/cluster --namespace openshift-machine-api
+----
+
+. Run the following command to edit the resource:
++
+[source,terminal]
+----
+$ oc edit controlplanemachineset.machine.openshift.io/cluster --namespace openshift-machine-api
+----
+
```
Comment (EmilienM) [inline modules/cpmso-openstack-with-az-config.adoc:89]: Don't we want to mention https://access.redhat.com/solutions/7013893 ?

File: modules/cpmso-openstack-with-az-config.adoc [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+[id="cpmso-openstack-with-az-config_{context}"]
+= Configuring {rh-openstack} clusters that have control plane machines with availability zones after an upgrade
+
+For some clusters that run on {rh-openstack-first} that you upgrade, you must manually update machine resources before you can use control plane machine sets if the following configurations are true:
+
+* You upgraded the cluster from {product-title} 4.13 to 4.14.
+ 
+* The cluster infrastructure is installer-provisioned.
+
+* Control plane machines were configured to use availability zones.
```
Comment (EmilienM) [inline modules/cpmso-openstack-with-az-config.adoc:15]: to make it clear:

```suggestion
* Control plane machines were distributed across multiple OpenStack Nova availability zones.
```

or

```suggestion
* Control plane machines were distributed across multiple compute availability zones.
```

File: modules/cpmso-openstack-without-az-config.adoc [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+[id="cpmso-openstack-ts-az-config_{context}"]
+= Configuring {rh-openstack} clusters that have machines with root volume availability zones after an upgrade
+
+For some clusters that run on {rh-openstack-first} that you upgrade, you must manually update machine resources before you can use control plane machine sets if the following configurations are true:
+
+* You upgraded the cluster from {product-title} 4.13 to 4.14.
+ 
+* The cluster infrastructure is installer-provisioned.
+
+* Machines were configured to use availability zones.
```
Comment (EmilienM) [inline modules/cpmso-openstack-without-az-config.adoc:15]: Same comment as https://github.com/openshift/openshift-docs/pull/66212/files#r1361012607

File: modules/cpmso-openstack-without-az-config.adoc [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+* Machines were configured to use root volumes for which availability zones were not defined.
```
Comment (EmilienM) [inline modules/cpmso-openstack-without-az-config.adoc:17]:
```suggestion
* Machines were configured to use root volumes for which OpenStack Cinder availability zones were not defined.
```

or

```suggestion
* Machines were configured to use root volumes for which block storage availability zones were not defined.
```

File: modules/cpmso-openstack-without-az-config.adoc [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+$ oc describe controlplanemachineset.machine.openshift.io/cluster --namespace openshift-machine-api
+----
+
+. Run the following command to edit the resource:
++
+[source,terminal]
+----
+$ oc edit controlplanemachineset.machine.openshift.io/cluster --namespace openshift-machine-api
+----
+
```
Comment (EmilienM) [inline modules/cpmso-openstack-without-az-config.adoc:96]: Same here, it'll be nice to link https://access.redhat.com/solutions/7024383

File: modules/cpmso-yaml-failure-domain-openstack.adoc [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+//
+// * machine_management/cpmso-configuration.adoc
+
+:_content-type: REFERENCE
+[id="cpmso-yaml-failure-domain-openstack_{context}"]
+= Sample {rh-openstack} failure domain configuration
+// TODO: Replace that link.
+The control plane machine set concept of a failure domain is analogous to existing {rh-openstack-first} concept of an link:https://docs.openstack.org/cinder/latest/admin/availability-zone-type.html[_Availability Zone (AZ)_]. The `ControlPlaneMachineSet` CR spreads control plane machines across multiple failure domains when possible. 
+
+When configuring {rh-openstack} failure domains in the control plane machine set, you must specify the availability zone name and the subnet to use.
```
Comment (EmilienM) [inline modules/cpmso-yaml-failure-domain-openstack.adoc:11]: the subnet?

Comment (maxwelldb) [reply modules/cpmso-yaml-failure-domain-openstack.adoc:11]: Not a fan of this whole description. Will replace.

File: modules/cpmso-openstack-without-az-config.adoc (hunk 3 above)
Comment (maxwelldb) [reply modules/cpmso-openstack-without-az-config.adoc:15]: @EmilienM This link isn't resolving to a particular file for me. Taking a guess, is it something like this?

```suggestion
* Machines were distributed across multiple availability zones.
```

Comment (EmilienM) [reply modules/cpmso-openstack-without-az-config.adoc:15]: yes, exactly

**Comment by maxwelldb** [comment]
@EmilienM Ready for another look, I think.

File: modules/cpmso-yaml-failure-domain-openstack.adoc [hunk 7]
```diff
@@ -0,0 +1,28 @@
+// Module included in the following assemblies:
+//
+// * machine_management/cpmso-configuration.adoc
+
+:_content-type: REFERENCE
+[id="cpmso-yaml-failure-domain-openstack_{context}"]
+= Sample {rh-openstack} failure domain configuration
+// TODO: Replace that link.
+The control plane machine set concept of a failure domain is analogous to existing {rh-openstack-first} concept of an link:https://docs.openstack.org/cinder/latest/admin/availability-zone-type.html[_Availability Zone (AZ)_]. The `ControlPlaneMachineSet` CR spreads control plane machines across multiple failure domains when possible.
```
Comment (EmilienM) [inline modules/cpmso-yaml-failure-domain-openstack.adoc:9]: I don't think that link should be used. I prefer this one: https://docs.openstack.org/nova/latest/admin/availability-zones.html

**Comment by EmilienM** [comment]
> @EmilienM Ready for another look, I think.

LGTM, I just made a suggestion.

File: modules/cpmso-yaml-failure-domain-openstack.adoc (hunk 7 above)
Comment (maxwelldb) [reply modules/cpmso-yaml-failure-domain-openstack.adoc:9]:
```suggestion
The control plane machine set concept of a failure domain is analogous to existing {rh-openstack-first} concept of an link:https://docs.openstack.org/nova/latest/admin/availability-zones.html[availability zone]. The `ControlPlaneMachineSet` CR spreads control plane machines across multiple failure domains when possible. 
```

File: modules/cpms-changing-openstack-flavor-type.adoc [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+[source,yaml]
+----
+providerSpec:
+  value:
+    ...
+    flavor: <1>
+----
+<1> Specify a larger {rh-openstack}  flavor type with the same base as the previous selection. For example, you can change `m6i.xlarge` to `m6i.2xlarge` or `m6i.4xlarge`.
+
+. Save your changes.
```
Comment (itzikb-redhat) [inline modules/cpms-changing-openstack-flavor-type.adoc:29]: Is it obvious that after the change the machines will be replaced with ones with a new flavor?

File: modules/cpms-changing-openstack-flavor-type.adoc [hunk 9]
```diff
@@ -0,0 +1,29 @@
+// Module included in the following assemblies:
+// * machine_management/control_plane_machine_management/cpmso-using.adoc
+
+:_content-type: PROCEDURE
+[id="cpms-changing-openstack-flavor-type_{context}"]
+= Changing the {rh-openstack} compute flavor by using a control plane machine set
```
Comment (itzikb-redhat) [inline modules/cpms-changing-openstack-flavor-type.adoc:6]: Are we using the flavor as an example?

File: modules/cpmso-openstack-ts-root-volume-azs.adoc [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+
+To understand why this procedure is necessary, see link: https://access.redhat.com/solutions/7013893[Solution #7024383].
+
+.Procedure
+
+. For all control plane machines, edit the provider spec for all control plane machines that match the environment. For example, to edit the machine `master-0`, enter the following command:
++
+[source,terminal]
+----
+$ oc edit machine/<cluster_id>-master-0 -n openshift-machine-api
```
Comment (itzikb-redhat) [inline modules/cpmso-openstack-ts-root-volume-azs.adoc:27]: The name can be different , right? maybe use a specific master name

File: modules/cpmso-openstack-with-az-config.adoc [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+. Run the following command to edit the resource:
++
+[source,terminal]
+----
+$ oc edit controlplanemachineset.machine.openshift.io/cluster --namespace openshift-machine-api
+----
+
+. For that resource, set the value of the `spec.state` property to `Active` to activate control plane machine sets for your cluster. 
+
+Your control plane is ready to be managed by the Cluster Control Plane Machine Set Operator.
```
Comment (itzikb-redhat) [inline modules/cpmso-openstack-with-az-config.adoc:94]: What about looking at the CPMSO logs to see that everything is fine? (@EmilienM wdyt?)

File: modules/cpms-changing-openstack-flavor-type.adoc [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+.Procedure
+
+. Edit the following line under the `providerSpec` field:
++
+[source,yaml]
+----
+providerSpec:
+  value:
+    ...
+    flavor: <1>
```
Comment (maxwelldb) [inline modules/cpms-changing-openstack-flavor-type.adoc:25]:
```suggestion
    flavor: m1.xlarge <1>
```

File: modules/cpms-changing-openstack-flavor-type.adoc (hunk 9 above)
Comment (maxwelldb) [reply modules/cpms-changing-openstack-flavor-type.adoc:6]: Not sure what you mean. Could you rephrase?

File: modules/cpmso-openstack-ts-root-volume-azs.adoc (hunk 10 above)
Comment (maxwelldb) [reply modules/cpmso-openstack-ts-root-volume-azs.adoc:27]: @itzikb-redhat Which part differs? The step assumes that the machine is named `master-0`. Is it not always prepended with the cluster ID?

File: modules/cpms-changing-openstack-flavor-type.adoc (hunk 9 above)
Comment (itzikb-redhat) [reply modules/cpms-changing-openstack-flavor-type.adoc:6]: I mean we can use other CPMS attributes and the flavor is just an example , right?

File: modules/cpms-changing-openstack-flavor-type.adoc [hunk 13]
```diff
@@ -22,8 +22,10 @@ In {rh-openstack}, flavors define the compute, memory, and storage capacity of c
 providerSpec:
   value:
     ...
-    flavor: <1>
+    flavor: m1.xlarge <1>
 ----
 <1> Specify a larger {rh-openstack}  flavor type with the same base as the previous selection. For example, you can change `m6i.xlarge` to `m6i.2xlarge` or `m6i.4xlarge`.
```
Comment (itzikb-redhat) [inline modules/cpms-changing-openstack-flavor-type.adoc:27]: @EmilienM Must it be a larger one ?

File: modules/cpms-changing-openstack-flavor-type.adoc (hunk 9 above)
Comment (maxwelldb) [reply modules/cpms-changing-openstack-flavor-type.adoc:6]: @itzikb-redhat It's what the previous writer included in their work--not 100% on it myself. I'm not sure how important the flavor task is specifically, nor if there are limitations on which attributes you can change. @EmilienM ?

File: modules/cpms-changing-openstack-flavor-type.adoc (hunk 13 above)
Comment (EmilienM) [reply modules/cpms-changing-openstack-flavor-type.adoc:27]: anything you want. The use case we should highlight is vertical scale. Where you use a bigger flavor or a smaller one is up to your needs.

File: modules/cpms-changing-openstack-flavor-type.adoc (hunk 9 above)
Comment (EmilienM) [reply modules/cpms-changing-openstack-flavor-type.adoc:6]: the flavor example is because CPMS helps with vertical scaling too. You can easily scale your control plane vertically by changing the flavors and obtain better perfs.

File: modules/cpms-changing-openstack-flavor-type.adoc [hunk 14]
```diff
@@ -0,0 +1,29 @@
+// Module included in the following assemblies:
+// * machine_management/control_plane_machine_management/cpmso-using.adoc
+
+:_content-type: PROCEDURE
+[id="cpms-changing-openstack-flavor-type_{context}"]
+= Changing the {rh-openstack} compute flavor by using a control plane machine set
+
+You can change the {rh-openstack-first} compute service (Nova) flavor that your control plane machines use by updating the specification in the control plane machine set custom resource.
+
+In {rh-openstack}, flavors define the compute, memory, and storage capacity of computing instances.
```
Comment (maxwelldb) [inline modules/cpms-changing-openstack-flavor-type.adoc:10]:
```suggestion
In {rh-openstack}, flavors define the compute, memory, and storage capacity of computing instances. By increasing or decreasing the flavor size, you can scale your control plane vertically.
```

File: modules/cpms-changing-openstack-flavor-type.adoc (hunk 13 above)
Comment (itzikb-redhat) [reply modules/cpms-changing-openstack-flavor-type.adoc:27]: So maybe it's good to state that because from what I understood from the text it should be larger.

File: modules/cpms-changing-openstack-flavor-type.adoc (hunk 9 above)
Comment (itzikb-redhat) [reply modules/cpms-changing-openstack-flavor-type.adoc:6]: Can we state with something like that - The scaling up is possible with CPMS by changing the flavor?

File: modules/cpms-changing-openstack-flavor-type.adoc (hunk 13 above)
Comment (maxwelldb) [reply modules/cpms-changing-openstack-flavor-type.adoc:27]: https://github.com/openshift/openshift-docs/pull/66212/commits/67aff27738a7b6429f828d88be92f452c9573d99

File: modules/cpms-changing-openstack-flavor-type.adoc (hunk 9 above)
Comment (maxwelldb) [reply modules/cpms-changing-openstack-flavor-type.adoc:6]: https://github.com/openshift/openshift-docs/pull/66212/commits/67aff27738a7b6429f828d88be92f452c9573d99

File: modules/cpms-changing-openstack-flavor-type.adoc (hunk 13 above)
Comment (maxwelldb) [reply modules/cpms-changing-openstack-flavor-type.adoc:27]:
```suggestion
<1> Specify a {rh-openstack}  flavor type that has the same base as the existing selection. For example, you can change `m6i.xlarge` to `m6i.2xlarge` or `m6i.4xlarge`. You can choose larger or smaller flavors depending on your vertical scaling needs.
```

Comment (maxwelldb) [reply modules/cpms-changing-openstack-flavor-type.adoc]: https://github.com/openshift/openshift-docs/pull/66212/commits/5126856bb971640d64e8f432ad85349188fa81f8

File: machine_management/control_plane_machine_management/cpmso-configuration.adoc [hunk 15]
```diff
@@ -33,6 +33,8 @@ The `<platform_provider_spec>` and `<platform_failure_domains>` sections of the
 
 * xref:../../machine_management/control_plane_machine_management/cpmso-configuration.adoc#cpmso-sample-yaml-vsphere_cpmso-configuration[Sample YAML snippets for configuring VMware vSphere clusters]
 
+* xref:../../machine_management/control_plane_machine_management/cpmso-configuration#cpmso-sample-yaml-openstack_cpmso-configuration[Sample YAML snippets for configuring {rh-openstack-first} clusters]
```
Comment (adellape) [inline machine_management/control_plane_machine_management/cpmso-configuration.adoc:36]:
```suggestion
* xref:../../machine_management/control_plane_machine_management/cpmso-configuration.adoc#cpmso-sample-yaml-openstack_cpmso-configuration[Sample YAML snippets for configuring {rh-openstack-first} clusters]
```

File: machine_management/control_plane_machine_management/cpmso-getting-started.adoc [hunk 16]
```diff
@@ -82,3 +87,4 @@ include::modules/cpmso-creating-cr.adoc[leveloffset=+1]
 * xref:../../machine_management/control_plane_machine_management/cpmso-configuration.adoc#cpmso-sample-yaml-azure_cpmso-configuration[Sample YAML for configuring Microsoft Azure clusters]
 * xref:../../machine_management/control_plane_machine_management/cpmso-configuration.adoc#cpmso-sample-yaml-nutanix_cpmso-configuration[Sample YAML for configuring Nutanix clusters]
 * xref:../../machine_management/control_plane_machine_management/cpmso-configuration.adoc#cpmso-sample-yaml-vsphere_cpmso-configuration[Sample YAML for configuring VMware vSphere clusters]
+* xref:../../machine_management/control_plane_machine_management/cpmso-configuration#cpmso-sample-yaml-openstack_cpmso-configuration[Sample YAML for configuring {rh-openStack-first} clusters]
```
Comment (adellape) [inline machine_management/control_plane_machine_management/cpmso-getting-started.adoc:90]:
```suggestion
* xref:../../machine_management/control_plane_machine_management/cpmso-configuration.adoc#cpmso-sample-yaml-openstack_cpmso-configuration[Sample YAML for configuring {rh-openStack-first} clusters]
```

File: machine_management/control_plane_machine_management/cpmso-resiliency.adoc [hunk 17]
```diff
@@ -23,6 +23,8 @@ include::modules/cpmso-failure-domains-provider.adoc[leveloffset=+2]
 
 * xref:../../machine_management/control_plane_machine_management/cpmso-configuration.adoc#cpmso-yaml-failure-domain-azure_cpmso-configuration[Sample Microsoft Azure failure domain configuration]
 
+* xref:../../machine_management/control_plane_machine_management/cpmso-configuration#cpmso-yaml-failure-domain-openstack_cpmso-configuration[Sample {rh-openstack-first} failure domain configuration]
```
Comment (adellape) [inline machine_management/control_plane_machine_management/cpmso-resiliency.adoc:26]:
```suggestion
* xref:../../machine_management/control_plane_machine_management/cpmso-configuration.adoc#cpmso-yaml-failure-domain-openstack_cpmso-configuration[Sample {rh-openstack-first} failure domain configuration]
```

File: machine_management/control_plane_machine_management/cpmso-troubleshooting.adoc [hunk 18]
```diff
@@ -28,4 +28,16 @@ include::modules/cpmso-ts-mhc-etcd-degraded.adoc[leveloffset=+1]
...
 [role="_additional-resources"]
 .Additional resources
-* xref:../../backup_and_restore/control_plane_backup_and_restore/disaster_recovery/scenario-2-restoring-cluster-state.adoc#dr-restoring-cluster-state[Restoring to a previous cluster state]
\ No newline at end of file
+* xref:../../backup_and_restore/control_plane_backup_and_restore/disaster_recovery/scenario-2-restoring-cluster-state.adoc#dr-restoring-cluster-state[Restoring to a previous cluster state]
+
+[id="cpmso-troubleshooting-shiftstack-upgrade_{context}"]
+== Upgrading clusters that run on {rh-openstack}
+
+For clusters that run on {rh-openstack-first} that you upgrade from {product-title} 4.13 to 4.14, you may have to perform post-upgrade tasks before you can use control plane machine sets.
```
Comment (adellape) [inline machine_management/control_plane_machine_management/cpmso-troubleshooting.adoc:36]: Suggest:
```suggestion
For clusters that run on {rh-openstack-first} that you upgrade from {product-title} 4.13 to 4.14, you might have to perform post-upgrade tasks before you can use control plane machine sets.
```

File: modules/cpms-changing-openstack-flavor-type.adoc [hunk 19]
```diff
@@ -0,0 +1,31 @@
...
+. Edit the following line under the `providerSpec` field:
++
+[source,yaml]
+----
+providerSpec:
+  value:
+    ...
+    flavor: m1.xlarge <1>
+----
+<1> Specify a {rh-openstack}  flavor type that has the same base as the existing selection. For example, you can change `m6i.xlarge` to `m6i.2xlarge` or `m6i.4xlarge`. You can choose larger or smaller flavors depending on your vertical scaling needs.
```
Comment (adellape) [inline modules/cpms-changing-openstack-flavor-type.adoc:27]: Extra space
```suggestion
<1> Specify a {rh-openstack} flavor type that has the same base as the existing selection. For example, you can change `m6i.xlarge` to `m6i.2xlarge` or `m6i.4xlarge`. You can choose larger or smaller flavors depending on your vertical scaling needs.
```

File: modules/cpms-changing-openstack-flavor-type.adoc [hunk 20]
```diff
@@ -0,0 +1,31 @@
...
+
+.Procedure
+
+. Edit the following line under the `providerSpec` field:
++
+[source,yaml]
+----
+providerSpec:
+  value:
+    ...
```
Comment (adellape) [inline modules/cpms-changing-openstack-flavor-type.adoc:24]: For [YAML snips](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#yaml-formatting):
```suggestion
# ...
```

File: modules/cpmso-openstack-ts-root-volume-azs.adoc [hunk 21]
```diff
@@ -0,0 +1,101 @@
...
+    - openshiftClusterID=refarch-lv7q9
+    trunk: true
+    userDataSecret:
+      name: master-user-data
+----
+<1> Set the zone name as this value.
++
+[NOTE]
+====
+If you edited or recreated machine resources after your initial cluster deployment, you may have to adapt these steps for your configuration.
```
Comment (adellape) [inline modules/cpmso-openstack-ts-root-volume-azs.adoc:80]:
```suggestion
If you edited or recreated machine resources after your initial cluster deployment, you might have to adapt these steps for your configuration. 
```

File: modules/cpmso-openstack-with-az-config.adoc [hunk 22]
```diff
@@ -0,0 +1,94 @@
...
+    - openshiftClusterID=refarch-lv7q9
+    trunk: true
+    userDataSecret:
+      name: master-user-data
+----
+<1> This value must match for machines `master-0`, `master-1`, and `master-3`.
++
+[NOTE]
+====
+If you edited or recreated machine resources after your initial cluster deployment, you may have to adapt these steps for your configuration.
```
Comment (adellape) [inline modules/cpmso-openstack-with-az-config.adoc:73]:
```suggestion
If you edited or recreated machine resources after your initial cluster deployment, you might have to adapt these steps for your configuration. 
```

File: modules/cpmso-openstack-with-az-config.adoc [hunk 23]
```diff
@@ -0,0 +1,94 @@
...
+
+For some clusters that run on {rh-openstack-first} that you upgrade, you must manually update machine resources before you can use control plane machine sets if the following configurations are true:
+
+* You upgraded the cluster from {product-title} 4.13 to 4.14.
+ 
+* The cluster infrastructure is installer-provisioned.
+
+* Control plane machines were distributed across multiple compute availability zones.
+
+To understand why this procedure is necessary, see link: https://access.redhat.com/solutions/7013893[Solution #7013893].
```
Comment (adellape) [inline modules/cpmso-openstack-with-az-config.adoc:17]:
```suggestion
To understand why this procedure is necessary, see link:https://access.redhat.com/solutions/7013893[Solution #7013893].
```

File: modules/cpmso-openstack-ts-root-volume-azs.adoc [hunk 24]
```diff
@@ -0,0 +1,101 @@
...
+
+* You upgraded the cluster from {product-title} 4.13 to 4.14.
+ 
+* The cluster infrastructure is installer-provisioned.
+
+* Machines were distributed across multiple availability zones.
+
+* Machines were configured to use root volumes for which block storage availability zones were not defined.
+
+To understand why this procedure is necessary, see link: https://access.redhat.com/solutions/7013893[Solution #7024383].
```
Comment (adellape) [inline modules/cpmso-openstack-ts-root-volume-azs.adoc:19]:
```suggestion
To understand why this procedure is necessary, see link:https://access.redhat.com/solutions/7013893[Solution #7024383].
```

**Review by adellape — COMMENTED** [review]
Only Nits In The Building 🕵️👍

File: modules/cpmso-openstack-ts-root-volume-azs.adoc [hunk 25]
```diff
@@ -0,0 +1,101 @@
...
+
+* You upgraded the cluster from {product-title} 4.13 to 4.14.
+ 
+* The cluster infrastructure is installer-provisioned.
+
+* Machines were distributed across multiple availability zones.
+
+* Machines were configured to use root volumes for which block storage availability zones were not defined.
+
+To understand why this procedure is necessary, see link:https://access.redhat.com/solutions/7013893[Solution #7024383].
```
Comment (EmilienM) [inline modules/cpmso-openstack-ts-root-volume-azs.adoc:19]: We need a release note for this. We have our first customer case now, they didn't follow the exact procedure.

File: modules/cpmso-openstack-with-az-config.adoc [hunk 26]
```diff
@@ -0,0 +1,94 @@
...
+
+For some clusters that run on {rh-openstack-first} that you upgrade, you must manually update machine resources before you can use control plane machine sets if the following configurations are true:
+
+* You upgraded the cluster from {product-title} 4.13 to 4.14.
+ 
+* The cluster infrastructure is installer-provisioned.
+
+* Control plane machines were distributed across multiple compute availability zones.
+
+To understand why this procedure is necessary, see link:https://access.redhat.com/solutions/7013893[Solution #7013893].
```
Comment (EmilienM) [inline modules/cpmso-openstack-with-az-config.adoc:17]: We need a release note for this. We have our first customer case now, they didn't follow the exact procedure.

File: modules/cpmso-openstack-ts-root-volume-azs.adoc (hunk 25 above)
Comment (maxwelldb) [reply modules/cpmso-openstack-ts-root-volume-azs.adoc:19]: Also looks like a copy-paste error in that link.
